pymeasure/display/widgets/multiplot_widget.py

```diff
@@ -0,0 +1,210 @@
+                self.plot_frame[i].change_x_axis(x_axis)
+        if y_axis is not None:
+            y_axis_label = y_axis
+            for idx, i in enumerate(range(self.num_plots)):
+                if type(y_axis) == list:
+                    y_axis_label = y_axis[idx]
+                self.columns_y[i].setCurrentIndex(self.columns_y[i].findText(y_axis_label))
+                self.plot_frame[i].change_y_axis(y_axis_label)
+
+    def _setup_ui(self):
+        self.columns_x_label = QtGui.QLabel(self)
+        self.columns_x_label.setMaximumSize(QtCore.QSize(45, 16777215))
+        self.columns_x_label.setText('X Axis:')
+        self.columns_y_label = []
+        for i in range(self.num_plots):
+            self.columns_y_label.append(QtGui.QLabel(self))
+            self.columns_y_label[i].setMaximumSize(QtCore.QSize(45, 16777215))
+            if i:
+                self.columns_y_label[i].setText('Y Axis ' + str(i + 1) + ':')
+            else:
+                self.columns_y_label[i].setText('Y Axis :')
+
+        self.columns_x = QtGui.QComboBox(self)
+        self.columns_y = []
+        for i in range(self.num_plots):
+            self.columns_y.append(QtGui.QComboBox(self))
+        for column in self.columns:
+            self.columns_x.addItem(column)
+            for i in range(self.num_plots):
+                self.columns_y[i].addItem(column)
+        self.columns_x.activated.connect(self.update_x_column)
+        for i in range(self.num_plots):
+            self.columns_y[i].activated.connect(partial(self.update_y_column, plot_idx=i))
+
+        self.plot_frame = []
+        for i in range(self.num_plots):
+            self.plot_frame.append(PlotFrame(
+                self.columns[0],
+                self.columns[1],
+                self.refresh_time,
+                self.check_status,
+                # show_label=(i == (self.num_plots - 1))
+            ))
+
+        self.updated = [self.plot_frame[i].updated for i in range(self.num_plots)]
+        self.plot = [self.plot_frame[i].plot for i in range(self.num_plots)]
+        self.columns_x.setCurrentIndex(0)
+        for i in range(self.num_plots):
+            self.columns_y[i].setCurrentIndex(1)
+
+    def _layout(self):
+        vbox = QtGui.QVBoxLayout(self)
+        vbox.setSpacing(0)
+
+        hbox = QtGui.QHBoxLayout()
+        hbox.setSpacing(10)
+        hbox.setContentsMargins(-1, 6, -1, 6)
+        hbox.addWidget(self.columns_x_label)
+        hbox.addWidget(self.columns_x)
+        for i in range(self.num_plots):
+            hbox.addWidget(self.columns_y_label[i])
+            hbox.addWidget(self.columns_y[i])
+
+        vbox.addLayout(hbox)
+        vbox.setSpacing(10)
+        for i in range(self.num_plots):
+            vbox.addWidget(self.plot_frame[i])
+        self.setLayout(vbox)
+
+    def new_curve(self, results, color=pg.intColor(0), **kwargs):
+        if 'pen' not in kwargs:
+            kwargs['pen'] = pg.mkPen(color=color, width=2)
```

Comment:
In line with the `PlotWidget` I would suggest to use `width=self.linewidth`.

Reply:
Fixed this issue